numpy/lib/polynomial.py

```diff
@@ -139,12 +139,11 @@ def poly(seq_of_zeros):
     if issubclass(a.dtype.type, NX.complexfloating):
         # if complex roots are all complex conjugates, the roots are real.
         roots = NX.asarray(seq_of_zeros, complex)
-        pos_roots = sort_complex(NX.compress(roots.imag > 0, roots))
-        neg_roots = NX.conjugate(sort_complex(
-                                        NX.compress(roots.imag < 0, roots)))
-        if (len(pos_roots) == len(neg_roots) and
-            NX.alltrue(neg_roots == pos_roots)):
-            a = a.real.copy()
+        pos_roots = NX.compress(roots.imag > 0, roots)
```

Comment:
Well, `.sort()` is in-place sorting and doesn't return anything, so `anything.sort() == anythingelse.sort()` is always true.

`np.all(np.sort(roots) == np.sort(roots.conjugate))`?

`sort_complex()` just calls `sort()` internally but always outputs a complex type?  Not sure if it's necessary, but just kept what was in the original code.